iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -2149,12 +2149,16 @@ public List<Partition> getPartitions(org.apache.hadoop.hive.ql.metadata.Table hm
   }
 
   public boolean isPartitioned(org.apache.hadoop.hive.ql.metadata.Table hmsTable) {
-    if ((hmsTable.getAsOfVersion() != null || hmsTable.getAsOfTimestamp() != null) &&
-            hasUndergonePartitionEvolution(hmsTable) ||
```

Comment:
I used `IcebergTableUtil.getTableSnapshot` to avoid duplicate knowledge of the snapshot configurations. I can move it back to the original style if we prefer it.